saviSimulation/savi/jason_processing/SAVI_Processing.java

```diff
@@ -0,0 +1,261 @@
+			image.translate(-image.width/2,-image.height/2);		
+				
+				// to adjust compassAngle to the image
+			image.rotate((float) ((float)uxv.getBehavior().getCompassAngle()+Math.PI/2));
+
+				//draw image
+			shape(image, wo.position.x+wo.pixels/2, wo.position.y+wo.pixels/2, wo.pixels, wo.pixels);
+			
+			//undo rotation and translation
+			image.rotate((-1)* (float) ((float)uxv.getBehavior().getCompassAngle()+Math.PI/2));
+			image.translate(image.width/2,image.height/2);		
+			
+			text(Double.toString(wo.position.z+(wo.pixels/2))+"\n"+Double.toString(wo.position.z-(wo.pixels/2)), wo.position.x, wo.position.y);
+			noFill();
+
+				//draw perception area
+			drawPerceptionArea(uxv);
+
+				//draw circle on objects perceived
+				double azimuth, elevation, perception, range, diam;
+				for(CameraPerception cpi : uxv.getBehavior().getVisibleItems()){
+					azimuth = cpi.getParameters().get(0);
+					elevation = cpi.getParameters().get(1);
+					range = cpi.getParameters().get(2);
+					diam = cpi.getParameters().get(3);
+					p1 = Geometry.absolutePositionFromPolar(azimuth, elevation, range, uxv.position, uxv.getBehavior().getCompassAngle());
+					//double horiz_angle = (uxv.getBehavior().getCompassAngle()+cpi.getParameters().get(0));// % 2* Math.PI;
+					//double cosv = Math.cos(horiz_angle);
+					//double sinv = Math.sin(horiz_angle);
+					//p1 = new PVector(Math.round(cosv*cpi.getParameters().get(2))+uxv.position.x, Math.round(sinv*cpi.getParameters().get(2))+uxv.position.y);
+					// draw circle over items visualized
+					ellipse(p1.x,p1.y, (float) diam*2, (float) diam*2);
+				}
+			}
+		else {
+		stroke(0);
+
+		shapeMode(PConstants.CENTER);
+		//simulator.shape(this.image, this.position.x, this.position.y,pixels,pixels);
+		//show height lower and upper
+		text(Double.toString(wo.position.z+(wo.pixels/2))+"\n"+Double.toString(wo.position.z-(wo.pixels/2)), wo.position.x, wo.position.y);
+		
+		if(visibleThings.contains(wo.getType())) {
+			shape(modelImages.get(wo.getType()), wo.position.x, wo.position.y,wo.pixels,wo.pixels);
+
+		}
+		}
+	}
+
+	public void drawPerceptionArea(UxV uxv) {
+		// calculate the projection of the perception distance on the ground (important for UAV)
+		float groundPerceptionDistance = (float) Math.sqrt(uxv.perceptionDistance*uxv.perceptionDistance - uxv.position.z*uxv.position.z);
+		arc(uxv.position.x, uxv.position.y, groundPerceptionDistance*2, groundPerceptionDistance*2, (float)(uxv.getBehavior().getCompassAngle()) - uxv.perceptionAngle/2, (float)(uxv.getBehavior().getCompassAngle()) + uxv.perceptionAngle/2);
+	}
+
+	
+	/*public static void main(String[] passedArgs) {
+		String[] appletArgs = new String[] { "savi.jason_processing.SAVI_Processing" };
+		if (passedArgs != null) {
+			PApplet.main(concat(appletArgs, passedArgs));
+		} else {
+			PApplet.main(appletArgs);
+		}
+	}*/
+	
+	public void mousePressed() {
+
+		if (playButton.MouseIsOver()) {
+			worldModel.pauseSimulation();
```

Comment:
I've noticed that the pause button does not change to the play button when the simulation is paused. I'm assuming it's a bug in this method.